### PR TITLE
Repair numpy pins

### PR DESCRIPTION
**Pull request by jdblischak**
**tl;dr** The numpy pinnings in `recipe/conda_build_config.yaml` were ignored because they were not propogated to the yaml files in `.ci_support/`. The cause was the use of `pin_compatible('numpy')` in the host dependencies

**question:** How important is it to support Python 3.7? conda-forge has officially [dropped support for Python 3.7](https://conda-forge.org/docs/user/announcements.html#dropping-python-3-7), and we could simplify the maintenance of this feedstock by also dropping it (so that we can simply rely on the upstream conda-forge pinnings)

I've long been confused by the updates to `recipe/conda_build_config.yaml` in a5c01af09e591a05cdab5712f1fea2e4ad164d6d since they seem unrelated to building htslib. I think I understand them now. In order to continue building for Python 3.7, we need to explicitly include it (plus all the other versions, since edits in `recipe/conda_build_config.yaml` are not additive, but completely replace the conda-forge pinnings). However, as I try to update the recipe for a Windows build, I keep getting numpy solver errors. And when I look at the currently successful Azure builds, the version of numpy installed is not consistent with the values in `recipe/conda_build_config.yaml`. I eventually figured out that this is because the `numpy` field is not progated to the yaml files in `.ci_support/`, and thus are completely ignored.

I figured out the solution by looking at the [arrow-cpp recipe](https://github.com/conda-forge/arrow-cpp-feedstock/blob/32a27bf36e839abf426c6ba1139d674accae51a7/recipe/meta.yaml). It only uses `pin_compatible('numpy')` in the run dependencies. Looking at the [conda docs](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#extra-jinja2-functions), it is only supposed to be used here:

> `pin_compatible('package_name', min_pin='x.x.x.x.x.x', max_pin='x', lower_bound=None, upper_bound=None)`: To be used as pin in run and/or test requirements.

This is also consistent with the [conda-forge advice for pinning numpy](https://conda-forge.org/docs/maintainer/knowledge_base.html#linking-numpy).

When I removed `pin_compatible()`, then `conda smithy rerender` adds numpy to the yaml files in `.ci_support/`. And during the builds, the exact pinned numpy versions are installed. However, in the tests, a more recent version of numpy is installed, which is consistent with the current bounds in the recipe:

https://github.com/TileDB-Inc/tiledb-vcf-feedstock/blob/63319b071061cdfc8ab3ae067f3291e31af4aed1/recipe/meta.yaml#L86-L87

While I was making this change, I also removed `pin_subpackage('libtiledbvcf', exact=True)` from the host dependencies since the conda docs also say this function is only supposed to be used in the run dependencies. This isn't as important as a change. I tested on my fork that it works without it, and to be extra cautious, I also pinned `{{ version }}`. Essentially, I'm trying to follow the official conda docs to avoid future problems. Note that arrow-cpp does use `pin_subpackage()` in their host deps, and I suspect that could be removed as well.

### CODE REVIEW

**Comment by Shelnutt2** [comment]
@jdblischak this is really great information! I did not have the time to dig into the pin issue when I forced things through with the upper bound previously. We definitely want to follow conda-forge guidelines.

As for python 3.7, we are going to have to address it soon, but for the next few weeks I'd like to keep supporting it.

**Comment by Shelnutt2** [comment]
> /AzurePipelines run

I _think_ I changed the permissions so you can invoke this now

**Comment by jdblischak** [comment]
I confirmed the numpy pinning has been repaired. Here's excerpts from the log for the [Linux build with libdeflate 1.17](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=30229&view=logs&j=bb1c2637-64c6-57bd-9ea6-93823b2df951&t=350df31b-3291-5209-0bb7-031395f0baa1) (you'll need to generate your own temporary link for the raw text if you want to re-run my command below). During the build stages, numpy 1.20 is used with all versions of Python except 3.10, which is paired with numpy 1.21. Then during the later test stages, they all use numpy 1.23.5 except Python 3.7 which uses numpy 1.21.


```sh
$ curl -s https://dev.azure.com/TileDB-Inc/836549eb-f74a-4986-a18f-7fbba6bbb5f0/_apis/build/builds/30229/logs/66 | grep -e 'python:' -e 'numpy:' -e 'START' -e 'END'
2023-02-15T22:26:04.9683142Z numpy:
2023-02-15T22:26:04.9686951Z   python:
2023-02-15T22:26:04.9688400Z python:
2023-02-15T22:31:17.6615590Z BUILD START: ['libtiledbvcf-0.22.2-h3156680_1.conda', 'tiledbvcf-py-0.22.2-py39h2fc66b3_1.conda', 'tiledbvcf-py-0.22.2-py38h839c6c6_1.conda', 'tiledbvcf-py-0.22.2-py310h9ef5c98_1.conda', 'tiledbvcf-py-0.22.2-py37h9479b99_1.conda']
2023-02-15T22:37:28.1530620Z     numpy:                      1.20.3-py39hd249d9e_2       conda-forge
2023-02-15T22:37:28.1535465Z     python:                     3.9.16-h2782a2a_0_cpython   conda-forge
2023-02-15T22:39:34.6856608Z     numpy:                      1.20.3-py38h8246c76_2       conda-forge
2023-02-15T22:39:34.6864060Z     python:                     3.8.16-he550d4f_1_cpython   conda-forge
2023-02-15T22:41:30.1402107Z     numpy:                      1.21.6-py310h45f3432_0      conda-forge
2023-02-15T22:41:30.1406366Z     python:                     3.10.9-he550d4f_0_cpython   conda-forge
2023-02-15T22:43:25.9426708Z     numpy:                      1.20.3-py37h3e96413_2       conda-forge
2023-02-15T22:43:25.9430657Z     python:                     3.7.12-hb7a2778_100_cpython conda-forge
2023-02-15T22:45:25.4177240Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/libtiledbvcf-0.22.2-h3156680_1.conda
2023-02-15T22:45:33.4052891Z TEST END: /home/conda/feedstock_root/build_artifacts/linux-64/libtiledbvcf-0.22.2-h3156680_1.conda
2023-02-15T22:45:33.4096028Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py39h2fc66b3_1.conda
2023-02-15T22:45:44.0153274Z     numpy:              1.23.5-py39h3d75532_0       conda-forge
2023-02-15T22:45:44.0158063Z     python:             3.9.16-h2782a2a_0_cpython   conda-forge
2023-02-15T22:45:59.1912705Z TEST END: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py39h2fc66b3_1.conda
2023-02-15T22:45:59.1958463Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py38h839c6c6_1.conda
2023-02-15T22:46:10.6771669Z     numpy:              1.23.5-py38h7042d01_0       conda-forge
2023-02-15T22:46:10.6777676Z     python:             3.8.16-he550d4f_1_cpython   conda-forge
2023-02-15T22:46:25.5940628Z TEST END: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py38h839c6c6_1.conda
2023-02-15T22:46:25.5984041Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py310h9ef5c98_1.conda
2023-02-15T22:46:36.1148206Z     numpy:              1.23.5-py310h53a5b5f_0      conda-forge
2023-02-15T22:46:36.1153188Z     python:             3.10.9-he550d4f_0_cpython   conda-forge
2023-02-15T22:46:52.5033934Z TEST END: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py310h9ef5c98_1.conda
2023-02-15T22:46:52.5086945Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py37h9479b99_1.conda
2023-02-15T22:47:24.1864098Z     numpy:              1.21.6-py37h976b520_0       conda-forge
2023-02-15T22:47:24.1867867Z     python:             3.7.12-hb7a2778_100_cpython conda-forge
2023-02-15T22:47:38.8259305Z TEST END: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py37h9479b99_1.conda
```

**Comment by jdblischak** [comment]
I also confirmed that the latest, locally built libtiledbvcf is installed during both the build and test stages:

```sh
~$ curl -s https://dev.azure.com/TileDB-Inc/836549eb-f74a-4986-a18f-7fbba6bbb5f0/_apis/build/builds/30229/logs/66 | grep -e 'libtiledbvcf:' -e 'START'
2023-02-15T22:31:17.6615590Z BUILD START: ['libtiledbvcf-0.22.2-h3156680_1.conda', 'tiledbvcf-py-0.22.2-py39h2fc66b3_1.conda', 'tiledbvcf-py-0.22.2-py38h839c6c6_1.conda', 'tiledbvcf-py-0.22.2-py310h9ef5c98_1.conda', 'tiledbvcf-py-0.22.2-py37h9479b99_1.conda']
2023-02-15T22:37:05.6030225Z Resource usage statistics from bundling libtiledbvcf:
2023-02-15T22:37:28.1526747Z     libtiledbvcf:               0.22.2-h3156680_1           local
2023-02-15T22:39:34.6850481Z     libtiledbvcf:               0.22.2-h3156680_1           local
2023-02-15T22:41:30.1398792Z     libtiledbvcf:               0.22.2-h3156680_1           local
2023-02-15T22:43:25.9423909Z     libtiledbvcf:               0.22.2-h3156680_1           local
2023-02-15T22:45:25.4177240Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/libtiledbvcf-0.22.2-h3156680_1.conda
2023-02-15T22:45:29.6538631Z     libtiledbvcf:    0.22.2-h3156680_1       local
2023-02-15T22:45:33.4041365Z Resource usage statistics from testing libtiledbvcf:
2023-02-15T22:45:33.4096028Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py39h2fc66b3_1.conda
2023-02-15T22:45:44.0149953Z     libtiledbvcf:       0.22.2-h3156680_1           local
2023-02-15T22:45:59.1958463Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py38h839c6c6_1.conda
2023-02-15T22:46:10.6767553Z     libtiledbvcf:       0.22.2-h3156680_1           local
2023-02-15T22:46:25.5984041Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py310h9ef5c98_1.conda
2023-02-15T22:46:36.1144472Z     libtiledbvcf:       0.22.2-h3156680_1           local
2023-02-15T22:46:52.5086945Z TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/tiledbvcf-py-0.22.2-py37h9479b99_1.conda
2023-02-15T22:47:24.1861481Z     libtiledbvcf:       0.22.2-h3156680_1           local
```

**Comment by jdblischak** [comment]
@Shelnutt2 @ihnorton Ready for review/merge

**Review by ihnorton — APPROVED** [review]
Thanks!
